src/collective/lazysizes/tests/test_transformer.py

```diff
@@ -88,3 +95,91 @@ def test_lazyload_tweet_no_script(self):
         element = lxml.html.fromstring(TWEET_NO_SCRIPT)
         # the transformer returns None (skip element)
         self.assertIsNone(self.transformer._lazyload_tweet(element))
+
+    def test_lazyload_availability(self):
+        """Test, if transform is available.
+        """
+        tpl = u'<html><body><div id="content">'\
+            + u'<img src="{0}"/>'\
+            + u'</div></body></html>'
+        url = u'http://example.com/foo.png'
+        html = tpl.format(url)
+
+        # Case 1: No transform for logged-in
+        result = self.transformer.transformIterable(html, 'utf-8')
+        self.assertEqual(result, None)
+
+        # LOGOUT for following tests
+        logout()
+
+        # Case 2: Successful transform for anonymous
+        result = self.transformer.transformIterable(html, 'utf-8')
+        img = result.tree.xpath('//img')[0]
+        self.assertEqual(img.attrib['data-src'], url)
+        self.assertEqual(img.attrib['src'], PLACEHOLDER)
+
+    def test_lazyload_custom_availability(self):
+        """Test custom availability.
+        """
+        tpl = u'<html><body><div id="content">'\
+            + u'<img src="{0}"/>'\
+            + u'</div></body></html>'
+        url = u'http://example.com/foo.png'
+        html = tpl.format(url)
+
+        # Instantiate transformer - we're going to modify the instance.
+        transformer = LazySizesTransform(None, self.layer['request'])
+
+        # We're logged in, right?
+        self.assertTrue(plone.api.user.is_anonymous() is False)
+
+        # By default, the transform isn't available for logged in users.
+        result = transformer.transformIterable(html, 'utf-8')
+        self.assertEqual(result, None)
+
+        # After providing a custom ``available`` method it is.
+        # NOTE: You should subclass and register the transform, not monkey
+        # patching like here.
+        transformer.available = lambda: True
+        result = transformer.transformIterable(html, 'utf-8')
+        img = result.tree.xpath('//img')[0]
+        self.assertEqual(img.attrib['data-src'], url)
+        self.assertEqual(img.attrib['src'], PLACEHOLDER)
+
+    def test_lazyload_blacklist(self):
+        """Test, if blacklisted elements are transformed.
```

Comment:
we must get rid of this tests if all cases are now covered here:

https://github.com/collective/collective.lazysizes/blob/master/src/collective/lazysizes/tests/test_lazysizes.py